### PR TITLE
fix: handles custom TLS certs in enrich_task_logs.py [DET-9803]

### PR DESCRIPTION
**Pull request by corban-beaird**
## Description
Fixes failures in CircleCI that result in logs not being shipped back to master as a result of silent TLS errors. 

This handles configuring the TLS cert for the `api.post()` call in `enrich_task_logs.py`; as well as, prints log shipping exception to `stderr` on the task container in the event that someone is listening.

## Test Plan
- Confirm: tests/cluster/test_logging.py passes
- Run: `det cmd run echo 'hello'` and confirm that 'hello' appears in the task log

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-9803

### CODE REVIEW

File: master/static/srv/enrich_task_logs.py [hunk 1]
```diff
@@ -12,13 +12,15 @@
...
+from determined.common.api import certs, errors
 
 # Example log message given below.
 # 2022-05-12 16:32:48,757:gc_checkpoints: [rank=0] INFO: Determined checkpoint GC, version 0.17.16-dev0
 # Below regex is used to extract the rank field from the log message.
 # Excluding empty spaces this regex matches rank in the above example as [rank=0]
-rank = re.compile("(?P<space1> ?)\[rank=(?P<rank_id>([0-9]+))\](?P<space2> ?)(?P<log>.*)")
+rank = re.compile(
+    "(?P<space1> ?)\[rank=(?P<rank_id>([0-9]+))\](?P<space2> ?)(?P<log>.*)"
+)
```
Comment (rb-determined-ai) [inline master/static/srv/enrich_task_logs.py:23]: something is wrong, this shouldn't be reflowed

File: master/static/srv/enrich_task_logs.py [hunk 2]
```diff
@@ -79,7 +81,9 @@ def run(self) -> None:
 
                     self.ship_queue.put(
                         {
-                            "timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
+                            "timestamp": datetime.datetime.now(
+                                datetime.timezone.utc
+                            ).isoformat(),
```
Comment (rb-determined-ai) [inline master/static/srv/enrich_task_logs.py:86]: ditto

File: master/static/srv/enrich_task_logs.py [hunk 3]
```diff
@@ -130,13 +136,16 @@ def ship(self) -> None:
...
+                api.post(self.master_url, "task-logs", self.logs, cert=self.cert)
                 self.logs = []
                 return
-            except errors.APIException as e:
+            except Exception as e:
                 tries += 1
                 if tries == max_tries:
-                    raise e
+                    print(
+                        "failed to ship logs: {e}\nLogs to Ship: {self.logs}}",
```
Comment (rb-determined-ai) [inline master/static/srv/enrich_task_logs.py:146]: omit `self.logs`, maybe emit a `len(self.logs)` instead.  This log shipper still ships all logs to stdout naturally, so if somebody sees this message it'll just be really hard to read, but not share a lot of extra info.

File: master/static/srv/enrich_task_logs.py [hunk 4]
```diff
@@ -173,7 +183,9 @@ def main(
     parser = argparse.ArgumentParser(
         description="read a stream and enrich it with the standard logging metadata"
     )
-    parser.add_argument("--stdtype", type=str, help="the stdtype of this stream", required=True)
+    parser.add_argument(
+        "--stdtype", type=str, help="the stdtype of this stream", required=True
+    )
```
Comment (rb-determined-ai) [inline master/static/srv/enrich_task_logs.py:188]: shouldn't be reflowed

File: master/static/srv/enrich_task_logs.py [hunk 5]
```diff
@@ -130,13 +136,16 @@ def ship(self) -> None:
         tries = 0
         while tries < max_tries:
             try:
-                api.post(self.master_url, "task-logs", self.logs)
+                api.post(self.master_url, "task-logs", self.logs, cert=self.cert)
                 self.logs = []
                 return
-            except errors.APIException as e:
+            except Exception as e:
```
Comment (rb-determined-ai) [inline master/static/srv/enrich_task_logs.py:142]: don't do `as e` if `e` is not being used.

Comment (corban-beaird) [reply master/static/srv/enrich_task_logs.py:142]: e is being used, it's being printed in ` print(
                        "failed to ship logs: {e}\nLogs to Ship: {ln(self.logs)}}",

Comment (rb-determined-ai) [reply master/static/srv/enrich_task_logs.py:138]: doh

**Review by rb-determined-ai — APPROVED** [review]

**Pull request merged**